hardware_interface/README.md

```diff
@@ -1,2 +1,132 @@
 robot agnostic hardware_interface package.
 This package will eventually be moved into its own repo.
+
+## Suggestions for `hardware_interface` implementation
```

Comment:
Should we call this “Best practices” instead of “Suggestions”?

Reply:
Best practices sound good also. I'll ping you after the last change.